### PR TITLE
adding duration tracking, and semantic logging capability to MVI errors

### PR DESCRIPTION
**Pull request by saneshark**
## Background
We need to have a better means of keeping track of various success or failure states for MVI errors, ideally in a way that could ultimately be visualized graphically. Semantic logger offers us just that, provided you have access to cloudwatch.

## Definition of Done
Replaces all the calls to `log_message_to_sentry` with a new method, `log_info_and_errors`, that first invokes the semantic_logger, and than optionally logs to sentry if an additional attribute indicating the error class is provided.

Additionally the whole thing is wrapped up in a `measure_info` call which should log information about successful MVI fetches as well as reporting the duration it took for the entire call to return.

### CODE REVIEW

**Comment by patrickvinograd** [comment]
Hey @saneshark this approach looks good. Additionally wanted to clue you in to the fact that we've successfully wired up Cloudwatch logs to AWS ElasticSearch service and from there to Grafana - this in the context of the API gateway work. For various reasons we want to do it selectively, not for every single log message, but with semantic logging it should be easy to generate a tag like `logmetrics: true` that developers can add to appropriate log entries and use that as the filter to ingest into ElasticSearch. (We haven't built that yet, but could do so). 

Then there's some work to define filters and queries on the data, but so far it's working for us, giving us ability to do queries by API consumer. If I can get around to deciding on that general filtering mechanism you could add it here for better visualization of this data, and then we could socialize it among the rest of the vets.gov engineers.

**Comment by saneshark** [comment]
@patrickvinograd that sounds great! I was thinking a little side project using d3.js and api calls might be a fun exercise, but it sounds like the solution you guys have cooking is something we could have working much sooner.

File: Gemfile.lock [hunk 1]
```diff
@@ -41,7 +41,6 @@ PATH
 
 GEM
   remote: https://rubygems.org/
-  remote: https://enterprise.contribsys.com/
```
Comment (jkassemi) [inline Gemfile.lock]: kill the Gemfile.lock changes since they'll kick sidekiq EE out of our build

File: lib/mvi/service.rb [hunk 2]
```diff
@@ -58,27 +60,37 @@ def find_profile(user)
...
+          log_info_and_errors('MVI Invalid Request', :error)
         end
       when MVI::Errors::FailedRequestError
-        log_message_to_sentry('MVI Failed Request', :error)
+        log_info_and_errors('MVI Failed Request', :error)
       end
     end
 
+    def log_info_and_errors(message, sentry_classification = nil)
+      Rails.logger.debug(message)
```
Comment (jkassemi) [inline lib/mvi/service.rb]: I think we're going to want info level messages here so we maintain these in the production logs

File: lib/mvi/service.rb [hunk 3]
```diff
@@ -58,27 +60,37 @@ def find_profile(user)
...
       when MVI::Errors::DuplicateRecords
-        log_message_to_sentry('MVI Duplicate Record', :warn)
+        log_info_and_errors('MVI Duplicate Record', :warn)
       when MVI::Errors::RecordNotFound
-        # Not going to log RecordNotFound
-        # NOTE: ICN based lookups do not return RecordNotFound. They return InvalidRequestError
-        # log_message_to_sentry('MVI Record Not Found', :warn)
-        nil
+        # Not going to log RecordNotFound to sentry, cloudwatch only.
+        log_info_and_errors('MVI Record Not Found')
```
Comment (jkassemi) [inline lib/mvi/service.rb]: we're going to have a little trouble tying the logging context to the previous measure_info call (we can't do a single query that matches back toward another entry in the cloudwatch console). 

Best bet is to duplicate that info in the log statement.

File: lib/mvi/service.rb (hunk 2 above)
Comment (saneshark) [reply lib/mvi/service.rb]: are you sure the debug ones don't trickle up? see the examples  below that I tailed from `log/test.log`

File: Gemfile.lock (hunk 1 above)
Comment (saneshark) [reply Gemfile.lock]: done

**Pull request merged**